lib/index.js

```diff
@@ -1,54 +1,57 @@
-import { observable, extendObservable, runInAction, reaction } from 'mobx'
+const { observable, extendObservable, runInAction, reaction } = require('mobx');
 
-function EventBus () {
-  this.metadata = observable({})
-  this.register = function (store) {
+exports.eventBus = (function () {
+  let metadata = observable({})
+  function registerToStore(store) {
     const { __subscriberMetadata } = store
     if (__subscriberMetadata) {
       __subscriberMetadata.forEach((s) => {
-        if (!this.metadata[s.topic]) {
+        if (!metadata[s.topic]) {
           const item = {}
           item[s.topic] = { seq: 0 }
-          extendObservable(this.metadata, item)
-          this.metadata[s.topic].event = {}
+          extendObservable(metadata, item)
+          metadata[s.topic].event = {}
         }
-        if (this.metadata[s.topic].seq > 0) {
+        if (metadata[s.topic].seq > 0) {
           const { cb, selector } = s
-          if (!selector || selector(this.metadata[s.topic])) {
-            runInAction(() => cb.apply(store, [this.metadata[s.topic]]))
+          if (!selector || selector(metadata[s.topic])) {
+            runInAction(() => cb.apply(store, [metadata[s.topic]]))
           }
         }
-        reaction(() => (this.metadata[s.topic].seq),
+        reaction(() => (metadata[s.topic].seq),
           (seq) => {
             if (seq > 0) {
               const { cb, selector } = s
-              if (!selector || selector(this.metadata[s.topic])) {
-                runInAction(() => cb.apply(store, [this.metadata[s.topic]]))
+              if (!selector || selector(metadata[s.topic])) {
+                runInAction(() => cb.apply(store, [metadata[s.topic]]))
               }
             }
           })
       })
     }
   }
 
-  this.post = function (topic, payload) {
+  function postEvent(topic, payload) {
     if (topic) {
-      if (!this.metadata[topic]) {
+      if (!metadata[topic]) {
         const item = {}
         item[topic] = { seq: 0 }
-        extendObservable(this.metadata, item)
+        extendObservable(metadata, item)
       }
       runInAction(() => {
-        this.metadata[topic].payload = payload
-        this.metadata[topic].seq = this.metadata[topic].seq + 1
+        metadata[topic].payload = payload
+        metadata[topic].seq = metadata[topic].seq + 1
       })
     }
   }
-}
 
-export const eventBus = new EventBus()
+  return {
+    register: registerToStore,
+    post: postEvent
+  }
+})();
 
-export function subscribe (topic, selector) {
+exports.subscribe = (topic, selector) => {

```

Comment:
No, idea behind module revealing pattern is slightly different. You are supposed to define object that includes everything exposed in a single place, and you separate it from definitions:

```
function xyz() {
}

functino abc() {
}

module.exports = {
  abc,
  xyz
}